bpf/ingress_node_firewall_kernel.c

```diff
@@ -183,8 +183,6 @@ ipv4_checkTuple(void *dataStart, void *dataEnd, __u32 ifId) {
             }
         }
 		bpf_printk("Packet didn't match any rule proto %d port %d", proto, bpf_ntohs(dstPort));
-        // we matched CIDR but we have no rules matching so drop
-        return SET_ACTIONRULE_RESPONSE(DENY, 0);
```

Comment:
This breaks some of the unit tests. Do you want me to fix them in a follow-up or do you want to do that?

~~~
[root@ovnkubernetes ingress-node-firewall]# make test
==== Generating DaemonSet manifest
hack/generate-daemon-manifest.sh
/root/development/ingress-node-firewall/bin/controller-gen rbac:roleName=manager-role crd webhook paths="./..." output:crd:artifacts:config=config/crd/bases
/root/development/ingress-node-firewall/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
go fmt ./...
go vet ./...
GOBIN=/root/development/ingress-node-firewall/bin
mkdir -p /root/development/ingress-node-firewall/testbin
test -f /root/development/ingress-node-firewall/testbin/setup-envtest.sh || curl -sSLo /root/development/ingress-node-firewall/testbin/setup-envtest.sh https://raw.githubusercontent.com/kubernetes-sigs/controller-runtime/v0.8.3/hack/setup-envtest.sh
source /root/development/ingress-node-firewall/testbin/setup-envtest.sh; fetch_envtest_tools /root/development/ingress-node-firewall/testbin; setup_envtest_env /root/development/ingress-node-firewall/testbin;
Using cached envtest tools from /root/development/ingress-node-firewall/testbin
setting up env vars
KUBEBUILDER_ASSETS="/root/.local/share/kubebuilder-envtest/k8s/1.24.1-linux-amd64" go test ./... -coverprofile cover.out
?   	github.com/openshift/ingress-node-firewall	[no test files]
?   	github.com/openshift/ingress-node-firewall/api/v1alpha1	[no test files]
?   	github.com/openshift/ingress-node-firewall/cmd	[no test files]
ok  	github.com/openshift/ingress-node-firewall/controllers	29.710s	coverage: 82.7% of statements
?   	github.com/openshift/ingress-node-firewall/pkg/apply	[no test files]
ok  	github.com/openshift/ingress-node-firewall/pkg/ebpf	0.093s	coverage: 0.0% of statements
{"level":"info","ts":1660763879.9748683,"logger":"syncIngressNodeFirewallResources","msg":"Running sync operation","ifaceIngressRules":null,"isDelete":false}
{"level":"info","ts":1660763879.9749808,"msg":"Creating a new eBPF firewall node controller"}
I0817 19:18:00.027869  765055 ingress_node_firewall_loader.go:321] Loading interfaces from pinned dir into memory
2022/08/17 19:18:00 EVENT_LOGGING_FILE env variable must be set
{"level":"info","ts":1660763880.0289068,"msg":"Comparing currently managed interfaces against list of XDP interfaces on system","e.managedInterfaces":{}}
{"level":"info","ts":1660763880.0289812,"msg":"Loading rules"}
I0817 19:18:00.029016  765055 ingress_node_firewall_loader.go:103] Ingress node firewall map Info: &{Type:LPMTrie id:230 KeySize:24 ValueSize:1200 MaxEntries:1024 Flags:1 Name:ingress_node_fi} with FD LPMTrie(ingress_node_firewall_table_map)#9
{"level":"info","ts":1660763882.1038787,"logger":"syncIngressNodeFirewallResources","msg":"Running sync operation","ifaceIngressRules":{"dummy0":[{"sourceCIDRs":["192.0.2.0/24"],"rules":[{"order":10,"protocolConfig":{"protocol":"TCP","tcp":{"ports":"12345"}},"action":"Deny"}]}]},"isDelete":false}
{"level":"info","ts":1660763882.103953,"msg":"Comparing currently managed interfaces against list of XDP interfaces on system","e.managedInterfaces":{}}
{"level":"info","ts":1660763882.103962,"msg":"Attaching firewall interface","intf":"dummy0"}
2022/08/17 19:18:02 Attached IngressNode Firewall program to iface "dummy0" (index 236)
{"level":"info","ts":1660763882.1043649,"msg":"Loading rules"}
I0817 19:18:02.104382  765055 ingress_node_firewall_loader.go:103] Ingress node firewall map Info: &{Type:LPMTrie id:230 KeySize:24 ValueSize:1200 MaxEntries:1024 Flags:1 Name:ingress_node_fi} with FD LPMTrie(ingress_node_firewall_table_map)#9
2022/08/17 19:18:02 Adding or updating ingress firewall rules for key {56 236 [192 0 2 0 0 0 0 0 0 0 0 0 0 0 0 0]}
--- FAIL: TestSyncInterfaceIngressRulesWithHTTP (8.54s)
    ebpfsyncer_test.go:1238: Checking if this test runs with sufficiently high privileges
    ebpfsyncer_test.go:1254: By populating the interface list
    ebpfsyncer_test.go:1268: By running a cleanup
    ebpfsyncer_test.go:1211: Deleting all XDP pins
    ebpfsyncer_test.go:1217: Deleting the test interfaces
    ebpfsyncer_test.go:1221: invalid argument
~~~

Reply:
fixed
```
I0817 16:02:07.973866   25213 ingress_node_firewall_loader.go:103] Ingress node firewall map Info: &{Type:LPMTrie id:138 KeySize:24 ValueSize:1200 MaxEntries:1024 Flags:1 Name:ingress_node_fi} with FD LPMTrie(ingress_node_firewall_table_map)#10
2022/08/17 16:02:07 Adding or updating ingress firewall rules for key {62 135 [192 0 2 4 0 0 0 0 0 0 0 0 0 0 0 0]}
2022/08/17 16:02:07 Adding or updating ingress firewall rules for key {62 133 [192 0 2 0 0 0 0 0 0 0 0 0 0 0 0 0]}
    ebpfsyncer_test.go:430: Waiting a little
    ebpfsyncer_test.go:435: TestSyncInterfaceIngressRulesWithHTTP(10): Trying to connect to 192.0.2.1:12346 (success should be false)
    ebpfsyncer_test.go:435: TestSyncInterfaceIngressRulesWithHTTP(10): Trying to connect to 192.0.2.5:12345 (success should be false)
    ebpfsyncer_test.go:435: TestSyncInterfaceIngressRulesWithHTTP(10): Trying to connect to 192.0.2.5:12346 (success should be true)
    ebpfsyncer_test.go:435: TestSyncInterfaceIngressRulesWithHTTP(10): Trying to connect to 192.0.2.1:12345 (success should be true)
    ebpfsyncer_test.go:1211: Deleting all XDP pins
    ebpfsyncer_test.go:1217: Deleting the test interfaces
    ebpfsyncer_test.go:1225: Deleting the netns
--- PASS: TestSyncInterfaceIngressRulesWithHTTP (42.58s)
```